src/SourceBuild/content/test/Microsoft.DotNet.SourceBuild.SmokeTests/BasicScenarioTests.cs

```diff
@@ -40,5 +42,31 @@ public static IEnumerable<TestScenario> GetScenarios()
             yield return new(nameof(BasicScenarioTests), language, DotNetTemplate.NUnit,    DotNetActions.Test);
             yield return new(nameof(BasicScenarioTests), language, DotNetTemplate.MSTest,   DotNetActions.Test);
         }
+
+        yield return new(nameof(BasicScenarioTests), DotNetLanguage.CSharp, DotNetTemplate.WebApp, DotNetActions.PublishSelfContained, VerifyRuntimePacksForSelfContained);
+    }
+
+    private static void VerifyRuntimePacksForSelfContained(string projectPath)
+    {   
+        string projNugetCachePath = Path.Combine(projectPath, "obj", "project.nuget.cache");
+        JsonNode? projNugetCache = JsonNode.Parse(File.ReadAllText(projNugetCachePath));
+
+        string failMessage = "Runtime packs were retrieved from NuGet instead of the SDK";
```

Comment:
Since this is only references once, consider inlining to improve the readability of the code.

Reply:
Sure thing, inlined